include/integration/simple_MC_integration.hpp

```diff
@@ -0,0 +1,283 @@
+typedef typename HPolytope<Point>::VT VT; 
+
+typedef const unsigned int Uint;  // Positive constant value for no of samples & dimensions
+enum volumetype {CB ,CG ,SOB}; // Volume type for polytope
+typedef typename std::vector<NT> Limit; // Standard way for user to use limits 
+// E.g. Limits LL{0.5, 1.5, 2.5} ; Limits UL{1.2, 1.8 , 2.8 } ;
+
+const Limit lt{0}; // To initialize non-initialized limits
+const Point pt{0}; // To initialize non-initialized points
+
+// To check if two n-dimensional points ensure valid limits in integration
+template
+<
+    typename Point = Point,
+    typename NT = NT
+>
+bool valid_limits(Point LL, Point UL) {
+    if (UL.dimension() == LL.dimension()) {
+        for (int i = 0; i<LL.dimension(); i++) {
+            if (LL[i] > UL[i]) {
+                std::cerr << "Invalid integration limits\n";
+                return false;
+            }
+        }
+        return true;
+    } else {
+        std::cerr << "Invalid integration limits\n";
+        return false;
+    }
+}
+
+// Initialize Limit Point
+template
+<
+    typename Point = Point,
+    typename NT = NT
+>
+Point init_limit(Limit L, Uint dim) {
+    Point pt(dim);
+    for (int i=0; i<dim; i++) {
+        pt.set_coord(i, L[i]);
+    }
+    return pt;
+}
+
+//Initialize to [-1,1]^n
+template
+<
+    typename Point = Point,
+    typename NT = NT
+>
+void initiate_unit_limits(Point& LL, Point& UL, int dim) {
+    LL.set_dimension(dim);
+    UL.set_dimension(dim);
+    LL.set_to_origin();
+    UL.set_to_origin();
+
+    for (int i=0 ; i<dim; i++) {
+        LL.set_coord(i, -1);
+        UL.set_coord(i, 1);
+    }
+}
+
+// Simple MC Integration Over Polytopes
+template 
+<
+    typename WalkType = BallWalk,
+    typename Polytope = HPOLYTOPE,
+    typename RNG = RandomNumberGenerator,
+    typename NT = NT,
+    typename Functor
+>
+NT simple_mc_polytope_integrate(Functor Fx, 
+                                Polytope &P, 
+                                Uint N = 10000, 
+                                volumetype voltype = SOB, 
+                                int walk_length = 1, 
+                                NT e = 0.1, 
+                                Point Origin = pt) 
+{
+
+    Uint dim = P.dimension();
+    // P.print();
+
+    // Check if ShiftPoint is shifted with accurate dimensions
+    if (Origin.dimension() == 0 && dim > 0) {
+        Origin.set_dimension(dim); 
+        Origin.set_to_origin();
+    } else if (Origin.dimension() != dim && dim > 0) {
+        std::cerr << "Polytope Dimension != Shiftpoint Dimension" << std::endl;
+        return -1;
+    }
+
+    // std::cout << "Origin.dimension() = " << Origin.dimension() << std::endl;
+    // std::cout << "P.dimension() = " << P.dimension() << std::endl;
+
+    // Volume calculation for HPolytope
+    NT volume;
+    
+    switch (voltype) {
+    case CB:     
+        volume = volume_cooling_balls <BallWalk, RNG, Polytope> (P, e, walk_length).second; 
```

Comment:
the walk parameter here should be part of the interface i.e. a new template e.g. `VolumeWalkType` that should be passed here instead of `BallWalk`

Reply:
SOB/CG are fine to good to go on this. 
What to do the CG Walktype? Cooling Gaussians seem to use GaussianFamily Walks only.